src/frontends/tensorflow_common/src/op/mul_no_nan.cpp

```diff
@@ -0,0 +1,40 @@
+#include "common_op_table.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert_like.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/select.hpp"
+
+using namespace std;
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+OutputVector translate_mul_no_nan_op(const NodeContext& node) {
+    default_op_checks(node, 2, {"MulNoNan"});
+
+    // first = x, second = y
+    auto x = node.get_input(0);
+    auto y = node.get_input(1);
+
+    // prepare zero constants of the same type as the inputs
+    auto zero = make_shared<v0::Constant>(element::f32, Shape{}, 0.0f)->output(0);
+    zero = make_shared<v1::ConvertLike>(zero, y);
```

Comment:
please use routine from utils:

```
auto const_zero = create_same_type_const_scalar<int32_t>(x, 0);
```

Do not forget to include `utils.hpp`

Reply:
@rkazants thanks for the review! Just updated the requested change.